ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/Animation.cs

```diff
@@ -270,12 +273,22 @@ private void ReadSegment(long frame, IKeyValueCollection segment, IKeyValueColle
                 {
                     case AnimDecoderType.CCompressedStaticFullVector3:
                     case AnimDecoderType.CCompressedFullVector3:
-                    case AnimDecoderType.CCompressedDeltaVector3:
                         outFrame.SetAttribute(boneNames[bone], channelAttribute, new Vector3(
                             containerReader.ReadSingle(),
                             containerReader.ReadSingle(),
                             containerReader.ReadSingle()));
                         break;
+                    case AnimDecoderType.CCompressedDeltaVector3:
+                        // Structure is | Bone 0 - Base Frame | Bone 1 - Base Frame | Bone 0 - Frame 0 | Bone 1 - Frame 0|
+                        if (containerReader.BaseStream.Position + (12 * numBones) < containerReader.BaseStream.Length)
+                        {
+                            containerReader.BaseStream.Position += 12 * numBones;
```

Comment:
So if I'm reading this right, it has a Vector3<float> for each bone stored (frame 0), and then the delta frames are Vector3<half>? Even the first frame has a delta of its own?

The only reason we really have a `containerReader.BaseStream.Length` check is because we don't fully understand decoders, so it doesn't overflow. But if this is correct implementation, this check is not needed here.

And you're sure the delta is always against the first frame and not the previous one? I'm not entirely sure that having a second binary reader (baseFrameContainerReader) is an elegant way of doing this.

Reply:
Whatever changes are required here to make it work correctly is for the best (I certainly think that swapping the switch/for is a good idea, probably can move each decoder into its own method).

The anim decoder is in dire need of proper implementation, it just about barely works currently. If you're willing to make it nicer here, I'm all for it.